### PR TITLE
Component/before-visit

### PR DESCRIPTION
**Pull request by Jood80**
### Content
- [x] NavBar #48 
- [x] Tabs #21 
- [x] Questions:
   - [x] What do you prefer for your visit to be
   - [x] What are you interested in viewing today
   - [x] Yes/No Question about Covid-19
- [x] Back button
- [x] Next button
---
- [x] style
- [x] Front-end validation 

### Testing
1. write these following commands down on your terminal :
```
$ pull origin component/before-visit
$ npm run client
```
2. browse http://localhost:3000/reservation.
3. the changes can be appear at Before visit page - second tab of the stepper-

### UI
image

### CODE REVIEW

File: client/src/components/layouts/BeforeVisit/index.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  const handleChangeCovid = (event) => {
+    setCovid(event.target.covid);
+  };
+  const handleChangeFurniture = (event) => {
+    setFurniture(event.target.furniture);
+  };
+
+  return (
+    <Fragment>
+      <Typography varient="h5" gutterBottom>
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/index.js]: variant

File: client/src/components/layouts/BeforeVisit/index.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        Before visit
+      </Typography>
+      <Typography varient="h6" gutterBottom>
+        Would you like to reserve a virtual appointment or in person?{' '}
+      </Typography>
+      <FormControl component="fieldset">
+        <RadioGroup
+          aria-label="meeting"
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/index.js]: Including these accessibility labels is excellent

File: client/src/components/layouts/BeforeVisit/index.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+import style from './style';
+
+const BeforeVisit = () => {
+  const classes = style();
+  const [meeting, setMeeting] = useState();
+  const [furniture, setFurniture] = useState();
+  const [covid, setCovid] = useState();
+
+  const handleChangeMeeting = (event) => {
+    setMeeting(event.target.meeting);
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/index.js]: I may be wrong but I think these might need to be `event.target.value`

File: client/src/components/layouts/BeforeVisit/index.js (hunk 1 above)
Comment (Jood80) [reply client/src/components/layouts/BeforeVisit/index.js]: Great Thanks!! I was trying to find out why the font is not responding to any changes I make. well here it is the reason :grin:

File: client/src/components/layouts/BeforeVisit/index.js (hunk 3 above)
Comment (Jood80) [reply client/src/components/layouts/BeforeVisit/index.js]: yeah, you're right.
at first, it works that way but I believe it's more accurate to be `.value`

File: client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js [hunk 4]
```diff
@@ -21,5 +21,20 @@ export default {
...
     },
+    appointmentType: {
+      name: 'meeting',
+      label: 'Meeting',
+      requiredErrorMsg: 'Meeting type is required',
+    },
+    product: {
+      name: 'product',
+      label: 'Product',
+      requiredErrorMsg: 'Choose one of products at least',
```
Comment (okjake) [inline client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js]: 'Choose at least one product'

File: client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
-      .matches(/^[0-9]{5}$/, 'Must be exactly 5 digits')
+      .matches(/^[0-9]{5}$/, "Must be exactly 5 digits")
       .required(`${zipCode.requiredErrorMsg}`),
   }),
+  Yup.object().shape({
+    [appointmentType.name]: Yup.string().required(
+      "Fill at least one"
+    ),
+    [product.name]: Yup.array().required("At least one is required"),
+    [covid19.name]: Yup.string().required("This Field is required"),
```
Comment (okjake) [inline client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js]: This field

File: client/src/components/layouts/BeforeVisit/BeforeVisit.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      <Typography variant="h6">
+        Would you like to reserve a virtual appointment or in person?
+      </Typography>
+      <MyRadio
+        data={appointmentOption}
+        name={appointmentType.name}
+        label={appointmentType.label}
+      />
+      <Typography variant="h6" gutterBottom>
+        What are you interested in viewing? select all that apply
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/BeforeVisit.jsx]: Select

File: client/src/components/layouts/BeforeVisit/BeforeVisit.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      </Typography>
+      <MyCheckbox
+        data={products}
+        name={product.name}
+        label={product.label}
+        className={classes.formControl}
+      />
+      <Typography variant="h6" gutterBottom>
+        Have you been in contact within the last 14 days with anyone who has
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/BeforeVisit.jsx]: any COVID symptoms?

File: client/src/components/layouts/BeforeVisit/Checkbox.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+              type="checkbox"
+              value={item.value}
+              as={Checkbox}
+              name={fieldName}
+            />
+            {item.label}
+          </label>
+        ))}
+      </Form>
+      <React.Fragment>{renderHelperText()}</React.Fragment>
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/Checkbox.jsx]: It doesn't really matter but this fragment isn't necessary

File: client/src/components/layouts/BeforeVisit/Radio.jsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        {data.map((item) => (
+          <FormControlLabel
+            key={item.label}
+            value={item.value}
+            control={<Radio />}
+            label={item.label}
+          />
+        ))}
+      </RadioGroup>
+      <React.Fragment>{renderHelperText()}</React.Fragment>
```
Comment (okjake) [inline client/src/components/layouts/BeforeVisit/Radio.jsx]: Same here

File: client/src/components/layouts/BeforeVisit/BeforeVisit.jsx (hunk 7 above)
Comment (Jood80) [reply client/src/components/layouts/BeforeVisit/BeforeVisit.jsx]: GUESS WHATT!! I was in hurry to push that before 12:00 am like cinderella's clock :smirk: so yess, you are absolutely right. This has to be corrected in more than spelling I believe, even the naming has to be more descriptive..

File: client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js [hunk 10]
```diff
@@ -1,22 +1,41 @@
...
+  }),
 ];
+
+// ).oneOf([true], 'Error'),
```
Comment (Rahaf-96) [inline client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js]: A comment! :open_mouth:

File: client/src/components/layouts/BeforeVisit/style.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+  },
+  formControl: {
+    // marginLeft: theme.spacing(3),
```
Comment (Rahaf-96) [inline client/src/components/layouts/BeforeVisit/style.js]: Another comment :open_mouth: HOW DARE YOU?

**Review by Rahaf-96 — CHANGES_REQUESTED** [review]

**Review by Rahaf-96 — APPROVED** [review]

**Review by tasneembhiri — APPROVED** [review]

**Review by sara219 — APPROVED** [review]

**Pull request merged**